pkg/api/plugins.go

```diff
@@ -440,8 +442,26 @@ func (hs *HTTPServer) InstallPlugin(c *contextmodel.ReqContext) response.Respons
 	}
 	pluginID := web.Params(c.Req)[":pluginId"]
 
+	repoCompatOpts := repo.NewCompatOpts(hs.Cfg.BuildVersion, runtime.GOOS, runtime.GOARCH)
+	jsonData, err := hs.pluginRepo.GetPluginJson(c.Req.Context(), pluginID, dto.Version, repoCompatOpts)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "Failed to fetch plugin json", err)
+	}
+	if jsonData.ExternalServiceRegistration != nil {
+		hs.log.Debug("check installer's permissions, plugin wants to register an external service")
+		hasAccess := accesscontrol.HasGlobalAccess(hs.AccessControl, hs.accesscontrolService, c)
+		// TODO (gamab) single_organization setup
+		evaluator := toEvalAll(jsonData.ExternalServiceRegistration.Permissions)
+		if !hasAccess(evaluator) {
+			c.JSON(http.StatusForbidden, map[string]string{
+				"title":   "Access denied",
+				"message": fmt.Sprintf("You'll need additional permissions to perform this action. Permissions needed: %s", evaluator.String()),
+			})
+		}
+	}
```

Comment:
I guess there is no better place than this to do this check. I thought of moving this to pkg/plugins/manager/installer.go to avoid the extra API request but that would imply to do the check after the plugin has been extracted so we would need additional cleanup.

I would at least move this code to a function.

Reply:
I had the same idea. Also it meant forwarding the user and moving the access control logic to the installer. If I recall correctly, the second point (move ac-ctl) is a "no! no!" as it implies partially breaking the isolation layer your squad is trying to build around the plugins packages.